Improve

SebastianRuffert opened the pull request.

SebastianRuffert committed:
Round numbers in NIMaxScreenshot throw error

Everytime a number of the format n.0000 is detected another screenshot needs to be taken, since it is highly unlikely the measurement sensor detects this kind of number. More often, it's due to an error

SebastianRuffert committed:
Template Data Output S001

Template for extracting data, at the Phelas Setup S001

SebastianRuffert committed:
Template for Tasklist

Templates can be used to start the measurement, by just entering a list with the corresponding input parameters

SebastianRuffert committed:
Minor Fix for Task Template

Changed default parameter from 'True' (str) to True (bool)

SebastianRuffert committed:
Data export Template: Code simplified

New function, which depending on chosen template get's the correct input. This way the template version can be implemented in custom code, instead of pasting the custom code and exchanging parameters.

JMoVS commented on MeasurementProgram.py:
```diff
@@ -140,29 +140,54 @@ def measure(self):
+                            "valid_options_steplength": 1e16}
+            answer = UserInput.ask_user_for_input(question)
+            meas.new_task(False,[2,0,answer["answer"],1.0])
+            #TODO One task is still missing, don't know what the input is
```
a bit unclear if this is still relevant?

SebastianRuffert replied:
Yes kind of, since I do not yet know if this corresponds to the wished procedure. I just need to see the specific progress of starting a measurement again

JMoVS commented on MeasurementProgram.py:
```diff
@@ -177,4 +202,4 @@ def version():
     version()
 MP = MeasurementProgram()
-MP.start()
+MP.start()
```
please don't remove the new line at the end of the file

JMoVS commented on DataStorage.py:
```diff
@@ -183,670 +204,705 @@ def get_task_id_from_task_list_index(index_in_task_list):
+        
+            
+    def _get_input(self,custom,question,template=[]):
+        #TODO
```
what is to do here?

SebastianRuffert committed:
Bug-Fix Template query

UserInput multi choice returns index not actual chosen input. Query must therefor check for index

SebastianRuffert committed:
Fixed Task List for S001 Measurement

Fixed Task List S001 for starting a new measurement, using the currently used procedure

SebastianRuffert committed:
Fixed Data Output template option

Fixed small bugs to make the template data output work. Changed the UserInput.confirm_warning function in order to not need to manually confirm warning

SebastianRuffert committed:
Included queue selection into template

Previously the queue selection was not included in the template option. This was fixed here

SebastianRuffert committed:
Fix Queue option in _get_input()

SebastianRuffert committed:
Revert "Fix Queue option in _get_input()"

This reverts commit 41d98b97628aaffe1a0fe7e7fb24f9cdd1f80388.

SebastianRuffert committed:
Revert "Included queue selection into template"

This reverts commit e7b50c257d83e7844bc77b7de659a196a35ded48.

SebastianRuffert committed:
Fix Queue Option

Included the queue selection into the template option

SebastianRuffert committed:
Excluded trigger seperation from template

Even in template case, trigger separation has to be entered manually

SebastianRuffert committed:
Fixed list output into integer

Multi indece question used where only first value was used. When implementing the template option the list was not converted into an integer and threw error.

JMoVS commented on DataStorage.py:
```diff
@@ -183,670 +197,848 @@ def get_task_id_from_task_list_index(index_in_task_list):
+            
+            #Step 5:
+            processing_log.append(time.strftime("%c") + ": Entering step 5")
+            indeces=0#Maybe 1 is correct, not sure
```
please remove comment

SebastianRuffert committed:
Small Fix

SebastianRuffert committed:
Small Fix

SebastianRuffert committed:
Fix

JMoVS commented:
thanks for the contribution!
